### PR TITLE
Add support for stop words

### PR DESCRIPTION
**Pull request by jasonbosco**
Related to #248.

Has some changes from #745, that needs to be merged first.

### CODE REVIEW

File: include/stop_word_index.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+private:
+
+    mutable std::shared_mutex mutex;
+    Store* store;
+    spp::sparse_hash_map<std::string, stop_word_t> stop_word_definitions;
+    spp::sparse_hash_map<uint64_t, std::vector<std::string>> stop_word_index;
+
+public:
+
+    static constexpr const char* COLLECTION_stop_word_PREFIX = "$CY";
```
Comment (kishorenc) [inline include/stop_word_index.h:54]: Should be `COLLECTION_STOP_WORD_PREFIX`

**Review by kishorenc — CHANGES_REQUESTED** [review]
a) Do we need to also ignore stop words during indexing? This will help in saving memory but offers less flexibility during querying.
b) Stop word index must be de-coupled from collection and be a separate top level resource that's attached to a collection.

**Comment by A4TIF** [comment]
Any progress on this? Would love to have this feature. Thanks

**Comment by jhwheeler** [comment]
@kishorenc What is the status of the stop words feature? Looks like it was merged, but I can't find any docs.

**Comment by kishorenc** [comment]
This has been merged to our current development branch and is available in `0.26.0.rc20`. Usage: https://gist.github.com/kishorenc/05593984e318e0ef5fcb80c300e7c9cf

**Comment by jhwheeler** [comment]
Great news, thank you for the update! Looking forward to this 🚀

**Comment by zehawki** [comment]
This is great, Ive been waiting for this for a long time. Notwithstanding https://typesense.org/docs/guide/migrating-from-algolia.html#key-features-in-algolia-not-in-typesense, "Server-side stop-words (can be implemented [client-side ](https://github.com/typesense/showcase-books-search/blob/b0b15dc88179566f85db21d2455c2d6c68668d5a/src/app.js#L186-L194))"

Would the stop words be configurable in this case?

**Pull request closed without merging**